sources/cassandra/toddl.go

```diff
@@ -0,0 +1,348 @@
+}
+
+func (tdi ToDdlImpl) ToSpannerType(conv *internal.Conv, spType string, srcType schema.Type, isPk bool) (ddl.Type, []internal.SchemaIssue) {
+	return tdi.typeMapper.GetSpannerType(srcType.Name, spType)
+}
+
+func (tdi ToDdlImpl) GetColumnAutoGen(conv *internal.Conv, autoGenCol ddl.AutoGenCol, colId string, tableId string) (*ddl.AutoGenCol, error) {
+	return &ddl.AutoGenCol{}, nil
+}
+
+func (tdi ToDdlImpl) GetTypeOption(srcTypeName string, spType ddl.Type) string {
+	return tdi.typeMapper.GetOption(srcTypeName, spType)
+}
+
+// CassandraDdlInfo encapsulates info about the ddl type, cassandra_type and issue
+type CassandraDdlInfo struct {
+	SpannerType ddl.Type
+	Option      string
+	Issues      []internal.SchemaIssue
+}
+
+// Static initialisation of base mapping
+var typeMappings = map[string][]CassandraDdlInfo{
+	"TINYINT": {
+		{
+			SpannerType: ddl.Type{Name: ddl.Int64},
+			Option:      "tinyint",
+			Issues:      []internal.SchemaIssue{internal.Widened},
+		},
+		{
+			SpannerType: ddl.Type{Name: ddl.String, Len: ddl.MaxLength},
+			Option:      "text",
+			Issues:      []internal.SchemaIssue{internal.Widened},
+		},
+	},
+	"SMALLINT": {
+		{
+			SpannerType: ddl.Type{Name: ddl.Int64},
+			Option:      "smallint",
+			Issues:      []internal.SchemaIssue{internal.Widened},
+		},
+		{
+			SpannerType: ddl.Type{Name: ddl.String, Len: ddl.MaxLength},
+			Option:      "text",
+			Issues:      []internal.SchemaIssue{internal.Widened},
+		},
+	},
+	"INT": {
+		{
+			SpannerType: ddl.Type{Name: ddl.Int64},
+			Option:      "int",
+			Issues:      []internal.SchemaIssue{internal.Widened},
+		},
+		{
+			SpannerType: ddl.Type{Name: ddl.String, Len: ddl.MaxLength},
+			Option:      "text",
+			Issues:      []internal.SchemaIssue{internal.Widened},
+		},
+	},
+	"BIGINT": {
+		{
+			SpannerType: ddl.Type{Name: ddl.Int64},
+			Option:      "bigint",
+			Issues:      nil,
+		},
+		{
+			SpannerType: ddl.Type{Name: ddl.String, Len: ddl.MaxLength},
+			Option:      "text",
+			Issues:      []internal.SchemaIssue{internal.Widened},
+		},
+	},
+	"FLOAT": {
+		{
+			SpannerType: ddl.Type{Name: ddl.Float32},
+			Option:      "float",
+			Issues:      nil,
+		},
+		{
+			SpannerType: ddl.Type{Name: ddl.Float64},
+			Option:      "double",
+			Issues:      []internal.SchemaIssue{internal.Widened},
+		},
+		{
+			SpannerType: ddl.Type{Name: ddl.String, Len: ddl.MaxLength},
+			Option:      "text",
+			Issues:      []internal.SchemaIssue{internal.Widened},
+		},
+	},
+	"DOUBLE": {
+		{
+			SpannerType: ddl.Type{Name: ddl.Float64},
+			Option:      "double",
+			Issues:      nil,
+		},
+		{
+			SpannerType: ddl.Type{Name: ddl.String, Len: ddl.MaxLength},
+			Option:      "text",
+			Issues:      []internal.SchemaIssue{internal.Widened},
+		},
+	},
+	"DECIMAL": {
+		// TODO: Generate appropriate SchemaIssue to warn of potential data loss
+		{
+			SpannerType: ddl.Type{Name: ddl.Numeric},
+			Option:      "decimal",
+			Issues:      nil,
```

Comment:
Could there be loss of precision between Cassandra Decimal and Spanner Numeric?
If yes, please consider adding an issue here.

Reply:
Just saw your TODO.